dags/common/sftp_service.py

```diff
@@ -16,33 +19,34 @@ def __init__(
         self.username = username
         self.password = password
         self.port = port
+        self.cnopts = pysftp.CnOpts()
+        self.cnopts.hostkeys = None
+        super().__init__(**self.__dict__)
+        self.logger = get_logger().bind(class_name=type(self).__name__)
         self.dir = dir
 
-    def list_files(self):
-        with self.__connect() as sftp:
-            return sftp.listdir()
-
-    def get_file(self, file):
-        with self.__connect() as sftp:
-            with sftp.open(file, "rb") as fl:
-                return BytesIO(fl.read())
-
-    def __connect(self):
-        cnopts = pysftp.CnOpts()
-        cnopts.hostkeys = None
-        conn = pysftp.Connection(
-            host=self.host,
-            username=self.username,
-            password=self.password,
-            port=self.port,
-            cnopts=cnopts,
-        )
-        if not conn.isdir(self.dir):
+    def __enter__(self):
+        # Bad practice, we need to fix it:
```

Comment:
should we solve it or keep it?

Reply:
let's fix the code it self and we will fix it in another PR since it's infra related issue